modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/XmlSdkImplicitConfig.java

```diff
@@ -0,0 +1,185 @@
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType.CONFIG;
+import static org.mule.runtime.api.functional.Either.right;
+import static org.mule.runtime.ast.api.ComponentGenerationInformation.EMPTY_GENERATION_INFO;
+import static org.mule.runtime.ast.api.ComponentMetadataAst.EMPTY_METADATA;
+import static org.mule.runtime.dsl.api.component.config.DefaultComponentLocation.from;
+import static org.mule.runtime.extension.internal.ast.MacroExpansionModuleModel.MODULE_CONFIG_GLOBAL_ELEMENT_NAME;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.component.ComponentIdentifier;
+import org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType;
+import org.mule.runtime.api.component.location.ComponentLocation;
+import org.mule.runtime.api.functional.Either;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterGroupModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterModel;
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.api.ComponentGenerationInformation;
+import org.mule.runtime.ast.api.ComponentMetadataAst;
+import org.mule.runtime.ast.api.ComponentParameterAst;
+import org.mule.runtime.ast.api.ParameterResolutionException;
+import org.mule.runtime.ast.api.util.BaseComponentAst;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * AST component that represent a configuration intended to be used when there is not a configuration on the application but there
+ * are modules from the XML Sdk that have XML Sdk properties with default values.
+ */
+public class XmlSdkImplicitConfig extends BaseComponentAst {
+
+  public static final String IMPLICIT_CONFIG_NAME_SUFFIX = "%s-xml-sdk-implicit-config";
+
+  private final ExtensionModel extensionModel;
+  private final String configName;
+  private final Set<ComponentParameterAst> componentParameterAsts = new HashSet<>();
+
+  public XmlSdkImplicitConfig(ExtensionModel extensionModel) {
+    this.extensionModel = extensionModel;
+    this.configName = format(IMPLICIT_CONFIG_NAME_SUFFIX, extensionModel.getName());
+
+    final List<ParameterModel> parameterModels = getModel(ParameterizedModel.class)
+        .map(ParameterizedModel::getAllParameterModels).orElse(emptyList());
+
+    for (ParameterModel parameterModel : parameterModels) {
+      String rawValue = parameterModel.getName().equals("name") ? configName : null;
+      ComponentParameterAst componentParameterAst = getComponentParameterAst(parameterModel, rawValue);
+      componentParameterAsts.add(componentParameterAst);
```

Comment:
why add everything with a nulle value instead of adding just the parameter for `name`?